cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py

```diff
@@ -586,6 +587,20 @@ def _create_block(request):
         boilerplate=request.json.get("boilerplate"),
     )
 
+    # If it contains library_content_key, the block is being imported from a v2 library
+    # so it needs to be synced with upstream block.
+    if upstream_ref := request.json.get("library_content_key"):
+        try:
+            # Set `created_block.upstream` so that we can try to sync and/or fetch.
+            # Note that, if this fails and we raise a 4XX, then we will not call modulstore().update_item,
+            # thus preserving the former value of `created_block.upstream`.
```

Comment:
```suggestion
            # Set `created_block.upstream` and then sync this with the upstream (library) version.
```
Why do we care about "preserving the former value of `created_block.upstream`" ? As I understand, `createed_block` is a new block and doesn't have any value for `upstream` yet until the line below.